src/commands/data/import/bulk.ts

```diff
@@ -0,0 +1,246 @@
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Successful records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return (data.numberRecordsProcessed - numberRecordsFailed).toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Failed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsFailed) {
+              return data.numberRecordsFailed.toString();
+            }
+          },
+        },
+      ],
+      postStagesBlock: [
+        {
+          label: 'Status',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) => data?.state,
+        },
+        {
+          label: 'Job Id',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) =>
+            data?.id &&
+            terminalLink(
+              data.id,
+              `${baseUrl}/lightning/setup/AsyncApiJobStatus/page?address=${encodeURIComponent(`/${data.id}`)}`
+            ),
+        },
+      ],
+    });
+
+    if (async) {
+      ms.goto('Creating ingest job');
+
+      const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+      ms.goto('Creating ingest job', job.getInfo());
+      ms.stop();
+
+      const cache = await BulkImportRequestCache.create();
+      await cache.createCacheEntryForRequest(job.id, conn.getUsername(), conn.getApiVersion());
+
+      this.log(messages.getMessage('export.resume', [job.id]));
+
+      return {
+        jobId: job.id,
+      };
+    }
+
+    // synchronous flow
+    const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+    ms.goto('Processing the job');
+
+    job.on('inProgress', (res: JobInfoV2) => {
+      ms.goto('Processing the job', res);
+    });
+
+    try {
+      await job.poll(5000, timeout.milliseconds);
+
+      const jobInfo = job.getInfo();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
+
+      if (jobInfo.numberRecordsFailed) {
+        ms.stop('failed');
+        // TODO: replace this msg to point to `sf data bulk results` when it's added (W-12408034)
+        throw messages.createError('error.failedRecordDetails', [
+          jobInfo.numberRecordsFailed,
+          conn.getUsername(),
+          job.id,
+        ]);
+      }
+
+      ms.stop();
+
+      return {
+        jobId: jobInfo.id,
+        processedRecords: jobInfo.numberRecordsProcessed,
+        successfulRecords: jobInfo.numberRecordsProcessed - (jobInfo.numberRecordsFailed ?? 0),
+        failedRecords: jobInfo.numberRecordsFailed,
```

Comment:
The API only gives us the total of records processed (that includes success/failure) and failed ones so we need calculate successful qty.
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_info.htm